scipy/stats/tests/test_morestats.py

```diff
@@ -728,10 +728,27 @@ def test_1d_input(self):
         assert_raises(ValueError, stats.levene, g1, x)
 
 
-class TestBinomP:
-    """Tests for stats.binom_test."""
+class TestBinomTestP:
+    """
+    Tests for stats.binomtest
+    """
+    @staticmethod
+    def binom_test_func(x, n=None, p=0.5, alternative='two-sided'):
+        # This processing of x and n is copied from deprecated stats.binom_test
+        x = np.atleast_1d(x).astype(np.int_)
+        if len(x) == 2:
+            n = x[1] + x[0]
+            x = x[0]
+        elif len(x) == 1:
+            x = x[0]
+            if n is None or n < x:
+                raise ValueError("n must be >= x")
+            n = np.int_(n)
+        else:
+            raise ValueError("Incorrect length for x.")
 
-    binom_test_func = staticmethod(stats.binom_test)
```

Comment:
I dont think we want to remove the test for `binom_test` until the deprecation is actually executed?

Reply:
I've been using this [PR](https://github.com/scipy/scipy/pull/16258) as an example. It felt OK to remove using deprecated methods. Could you please elaborate on the profits of keeping deprecated method tests?

Reply:
Updated PR, could you please take a look again? @j-bowhay

Reply:
Agreed no test should be removed until we remove the function itself.

Reply:
OK, I've returned the tests back in the 2nd commit now :)

Reply:
Looks better thanks